exporter/elasticsearchexporter/integrationtest/exporter_test.go

```diff
@@ -20,27 +20,41 @@ func TestExporter(t *testing.T) {
 	for _, eventType := range []string{"logs", "traces"} {
 		for _, tc := range []struct {
 			name string
+
+			// batcherEnabled enables/disables the batch sender. If this is
+			// nil, then the exporter buffers data itself (legacy behavior),
+			// whereas if it is non-nil then the exporter will not perform
+			// any buffering itself.
+			batcherEnabled *bool
+
 			// restartCollector restarts the OTEL collector. Restarting
 			// the collector allows durability testing of the ES exporter
 			// based on the OTEL config used for testing.
 			restartCollector bool
 			mockESFailure    bool
 		}{
 			{name: "basic"},
+			{name: "batcher_enabled", batcherEnabled: ptrTo(true)},
```

Comment:
non blocker: should we also test with 2 cases, `restartCollector: true` and `mockESFailure: true`?

Reply:
I've added test cases for `mockESFailure: true`, but not `restartCollector: true`. IIANM the latter would require adding persistent queuing to the integration tests, which I'd rather leave to a follow up if it's OK with you.